docs/examples/custom_auth.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function
+from twisted.internet.endpoints import serverFromString
+from twisted.internet.task import react
+from twisted.internet.defer import inlineCallbacks, maybeDeferred
+from twisted.web.resource import Resource
+from twisted.web.server import Site
+from twisted.web.http_headers import Headers
+from twisted.web.http import FORBIDDEN
+
+import treq
+
+
+class SillyAuthResource(Resource):
+    """
+    A resource that uses a silly, header-based authentication
```

Comment:
Thinking from the perspective of a non-native English speaker for a moment - the "silliness" here is not obvious, so perhaps it would be better to use a word like "custom"?